src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -645,15 +645,15 @@ discover_dfuse(void)
 			dfs_list[num_dfs].dfs_dir_hash = NULL;
 			dfs_list[num_dfs].len_fs_root  = strnlen(fs_entry->mnt_dir, DFS_MAX_PATH);
 			if (dfs_list[num_dfs].len_fs_root >= DFS_MAX_PATH) {
-				D_WARN("mnt_dir[] is too long! Skip this entry.");
+				D_DEBUG(DB_ANY, "mnt_dir[] is too long! Skip this entry.\n");
 				continue;
 			}
 			atomic_store_relaxed(&dfs_list[num_dfs].inited, 0);
 			dfs_list[num_dfs].pool         = NULL;
 			pt_end = stpncpy(dfs_list[num_dfs].fs_root, fs_entry->mnt_dir,
 					 DFS_MAX_PATH - 1);
 			if ((long int)(pt_end - dfs_list[num_dfs].fs_root) >= (DFS_MAX_PATH - 1)) {
-				D_WARN("fs_root[] is too long. Skip this entry.");
+				D_DEBUG(DB_ANY, "fs_root[] is too long. Skip this entry.\n");
 				continue;
```

Comment:
This is the same check as above on L646 I believe?  In which case you can D_ASSERT() here as it shouldn't ever fail.  Alternatively this would be a great place to move to using a dynamically allocated string rather than a hardcoded buffer, particularly as this is in bss memory 8 times.

Reply:
Right. I just removed the check here since we checked previously. strncpy() is used now. I also use dynamic allocation for dfs_list[].fs_root now. Thank you!